site/docs/intro/under-the-hood.md

```diff
@@ -2,5 +2,54 @@
 sidebar_position: 4
 title: Under the hood
 ---
+## Provisioning preview environments
 
-# Under the hood
+When provisioning a new environment using the `up` command, `preevy` does the following:
+- Reads for default configuration and relevant keys from the current profile store.
+- Calculates environment name based on git branch or uses the `--id` flag.
+- Uses the local aws configuration to provision a new Lightsail VM.
+- Reads SSH keypair from profile to access the VM, if necessary, generate a new one.
+- Connects to the VM using SSH and setup Docker.
+- Reads the compose file and copies local volume mounts to the VM.
+- Augments the compose deployment with a helper service, `tunnel-agent`, responsible for connecting to the [tunnel server](/tunnel-server/overview.md).
+- Runs the application using Docker Compose with `--build` while using the local build context.
+- The `tunnel-agent` is inspecting the network configuration of all deployed services and create a tunnel for each service.
+- Fetch the urls from tunnel-agent and output them to the end user.
+
+## Profile configuration
+
+Preevy profile provide a mechanism for storing and sharing configuration and keys between different machines (multiple developers and/or CI Jobs).
```

Comment:
```suggestion
The preevy configuration and state can be shared using the Preevy profiles store. This allows sharing of environments between different CI jobs, or different developers, without repeating the setup procedures. Because URLs for an environments are based on the Preevy state, it also means that URLs are stable between CI runs.
```

Reply:
updated